src/test/java/com/atlassian/db/replica/api/ThrottledCacheTest.java

```diff
@@ -0,0 +1,75 @@
+package com.atlassian.db.replica.api;
+
+import org.junit.Test;
+
+import java.util.ConcurrentModificationException;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class ThrottledCacheTest {
+    private final ExecutorService executor = Executors.newFixedThreadPool(10);
```

Comment:
```suggestion
    private final ExecutorService executor = Executors.newSingleThreadExecutor()
```

Reply:
It's not really concurrent ;)

Reply:
that's my point it can be simplified,

Reply:
Added something without arbitraty number. Async in a new thread is easier to test than async on _maybe_ new thread.